packages/iov-bns/src/util.ts

```diff
@@ -21,9 +22,15 @@ import { Sha256 } from "@iov/crypto";
 import { Bech32, Encoding } from "@iov/encoding";
 import { QueryString } from "@iov/tendermint-rpc";
 
+const bnsMainnetChainId = "PLEASE_INSERT_HERE_WHEN_GENESIS_EXISTS" as ChainId;
```

Comment:
Ahhhh.... we hardcode one `iov` chainId, rest are `tiov`.
Decent decision.

What do you think of the idea of allowing the client to pass in this prefix in the constructor of the bnsCodec. Eg. make the bnsCodec itself no longer a singleton, but a class with constructor (needing a prefix arg)?

Same way we configure the BcpConnection with some nodes, we can also configure the codec with some parameters... these come from client-side config, or later directly from bns... no need to hardcode something that can be controlled externally, if we can do so safely

(this proposal of using as arg can be a separate PR. the current format is fine and can be merged like this, more me thinking of an additional refinement, if you like it, make a new issue)

Reply:
What I implemented here (1 mainnet vs. other = testnet) is how we defined the prefixed "iov"/"tiov", so this is the simplest implementation that serves our needs. I would not create unnecessary flexibility as bnsCodec's job is to implement BNS rules.

I like codec to be stateless and would not change this unless we have a very good reason to.